src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -197,7 +205,11 @@ private CloseableIterator<ScoredRowId> searchInternal(AbstractBounds<PartitionPo
                 for (long i = minSSTableRowId; i <= maxSSTableRowId; i++)
                     segmentRowIds.add(metadata.toSegmentRowId(i));
 
-                return bruteForceOrdering(queryVector, segmentRowIds, limit, topK, threshold);
+                // When we have a threshold, we only need to filter the results, not order them.
```

Comment:
... because it means we're evaluating a boolean predicate in the SAI pipeline that wants to collate by PK